pkg/utils/configs/config.go

```diff
@@ -51,17 +52,23 @@ func SetQdrouterdDefaults(m *v1alpha1.Qdrouterd) bool {
 	if !isDefaultSslProfileDefined(m) && isDefaultSslProfileUsed(m) {
 		m.Spec.SslProfiles = append(m.Spec.SslProfiles, v1alpha1.SslProfile{
 			Name:        "default",
-			Credentials: m.Name + "-cert",
+			Credentials: m.Name + "-ss-tls",
 		})
-		requestCert = true
+		requestSSCert = true
 	}
-	for _, profile := range m.Spec.SslProfiles {
-		if profile.Credentials == "" {
-			profile.Credentials = m.Name + "-cert"
-			requestCert = true
+	for i := range m.Spec.SslProfiles {
+		if m.Spec.SslProfiles[i].Credentials == "" {
+			if m.Spec.SslProfiles[i].RequireClientCerts {
+				m.Spec.SslProfiles[i].Credentials = m.Name + "-ca-tls"
+				m.Spec.SslProfiles[i].CaCert = m.Name + "-ca-tls"
+				requestCACert = true
+			} else {
+				m.Spec.SslProfiles[i].Credentials = m.Name + "-ss-tls"
+				requestSSCert = true
+			}
 		}
 	}
-	return requestCert
+	return requestSSCert, requestCACert
```

Comment:
I may be misunderstanding the patch, but I think the issuer used should always be a CA issuer. It may be a CA provided by the user via the Issuer field, or it will be a self-signed CA (i.e. a CA certificate issued by the self signed issuer).

A separate decision is then whether we need a further CA certificate to be generated for the purposes of client authentication or whether we just use the same CA as used to issue the server certificate. This question I think is related to whether we want unique certificates for different ssl profiles. I would tentatively argue that we do; if the user has specified distinct ssl profiles, I would think they would expect distinct cert and CA if applicable. 

The case of inter-router certs for example, is one where you would want a CA different from that used to verify client connections, to ensure that only routers with access to the inter-router cert can be accepted as inter-router connections.